### PR TITLE
adding presence tracking to pusher

### PR DESCRIPTION
**Pull request by nolman**
CC @trevorcreech @timcour @codeferret1 @azirbel 

This PR adds presence tracking to pusher. This change is actually pretty small most of the changes are resulting from the inline here docs that came out of a generated file.

You can find more context on Presence tracking in phoenix [here](http://work.stevegrossi.com/2016/07/11/building-a-chat-app-with-elixir-and-phoenix-presence/) including the some of the changes we are going to need to make on the client side in web.

This change should be safe to go out as is as doesn't change any existing behavior, it will just be a feature we don't use yet on the frontend side.

### CODE REVIEW

File: web/channels/event_channel.ex [hunk 1]
```diff
@@ -24,6 +26,18 @@ defmodule Pusher.EventChannel do
     { :error, :authentication_required }
   end
 
+  def handle_info(:after_join, socket) do
+    push socket, "presence_state", Presence.list(socket)
+    claims = current_claims(socket)
+    is_map = Pusher.IsAHelper.is_a_map?(claims["sub"])
+    if(is_map && Map.has_key?(claims["sub"], "email")) do
+      {:ok, _} = Presence.track(socket, current_resource(socket)["email"], %{
```
Comment (trevorcreech) [inline web/channels/event_channel.ex]: should be `claims["sub"]["email"]

File: lib/pusher/is_a_helper.ex [hunk 2]
```diff
@@ -0,0 +1,9 @@
+defmodule Pusher.IsAHelper do
+  def is_a_map?(%{}) do
+    true
+  end
+
+  def is_a_map?(_) do
+    false
+  end
+end
```
Comment (trevorcreech) [inline lib/pusher/is_a_helper.ex]: maybe check with @hassox to see if there's a more elixir-y way to do this?

**Comment by trevorcreech** [comment]
Also need to update opendoor homebrew repo to push this out

File: web/channels/event_channel.ex (hunk 1 above)
Comment (codeferret1) [inline web/channels/event_channel.ex]: I think we're checking if the claims["sub"] hash has the key "email"?

**Pull request merged**